### ISSUE TITLE
Consistent docstring & documentation terminology

### ISSUE THREAD

**Issue by cczhu** [mhvk/baseband#116]
Documentation for parameters shared between formats should have consistent (identical, if appropriate) descriptions, and should use terminology that's consistent with each other and with the documentation.  Proposed common terms and descriptions are (currently an incomplete list):

Terms (maybe we need a glossary?):
- **channel**: may mean frequency sub-band channels or Fourier channels, depending on the data set.
- **frame**: a block of data, or payload, accompanied by a header.
- **header**: metadata accompanying a data frame.
- **payload**: the data within a data frame.
- **sample**: an individual data point.
- **complete sample**: samples from all threads, polarization, channels, etc. for one point in time.
- **sample rate**: rate of complete samples.
- **component**: individual threads, polarizations or channels of the complete sample.
- **elementary sample** (maybe change to **atomic sample**?): the smallest possible subdivision of a complete sample, i.e. the real/imaginary part of one component of a complete sample.
- **elementary sample rate**: rate of elementary samples, typically used only when describing the rate at which data is being read from or written to disk, or from the internet.
- **stream**: timeseries of complete samples; may refer to all of, or a subsection of, the dataset.
- **subset**: a subset of a complete sample, in particular one defined by the user for selective decoding.

`open` parameters when reading (each format may have specific defaults or features that are additionally described):
```
    fh_raw : 
        File handle of the raw <FORMAT> stream
    sample_rate : `~astropy.units.Quantity`
        Number of complete samples per second (ie. the sampling rate of each <COMPONENTS>).
    ref_time : `~astropy.time.Time`, or None, optional
        Reference time within <TIME FRAME> of the start time of the observations.
    samples_per_frame : int
        Number of complete samples per frame.
    nthread : int
        Number of threads in a complete sample.
    nchan : int
        Number of channels in a complete sample.
    complex_data : bool
        Whether the data is complex.
    bps : int
        Bits per elementary sample (e.g., the real or imaginary part of one
        component of the complete sample).
    subset: indexing object, or tuple of objects
        Specific components of the complete sample to decode.  <SOME FORMAT SPECIFIC INSTRUCTIONS.>
    squeeze : bool
        If `True` (default), remove any dimensions of length unity from
        decoded data.
```

`open` parameters when writing:
```
    raw :
        Which will write filled sets of frames to storage.
    header : :class:`~baseband.vdif.VDIFHeader`
        Header for the first frame, holding time information, etc.
    squeeze : bool
        If `True` (default), ``write`` accepts squeezed arrays as input,
        and adds channel and thread dimensions if they have length unity.
    **kwargs
        If no header is given, an attempt is made to construct the header from
        these.  For a standard header, the following suffices.
```

**Comment by mhvk** [comment]
Yes, good idea.

For reading, the arguments can be separated into things that cannot be inferred from the header, and things that help determine what exactly should be read. Likely a sensible logic is for writing to put those things that cannot be stored in the header right after `header` and before `squeeze`.

**Comment by cczhu** [comment]
This issue is tied together with #112, and maybe I should have just appended a comment to that one, but felt like it's quite important we get a universal terminology correct.  What are you thoughts on the terms?

**Comment by mhvk** [comment]
I think the main part of this has been addressed by #121, and what remains really belongs to #112 (where I now copied the example open docstring). Closing this one.
